lib/rules/no-unused-expressions.js

```diff
@@ -211,7 +216,9 @@ module.exports = {
 			ExpressionStatement(node) {
 				if (
 					Checker.isDisallowed(node.expression) &&
-					!isDirective(node)
+					(ignoreDirectives
+						? !isDirective(node)
+						: !astUtils.isDirective(node))
```

Comment:
```suggestion
					!astUtils.isDirective(node) &&
					!(ignoreDirectives && isDirective(node))
```

I think we can always check if the statement was parsed as a directive, as that should be the most reliable check and it isn't costly.